code/game/objects/structures/grille.dm

```diff
@@ -10,208 +10,215 @@
-				if( x == user.x )
-					if( y > user.y )
-						dir_to_set = 2
+			if((x == user.x) || (y == user.y)) //Only supposed to work for cardinal directions, aka can't lay windows in diagonal directions
+				if(x == user.x) //User is on the same vertical plane
+					if(y > user.y)
+						dir_to_set = 2 //User is laying from the bottom
 					else
-						dir_to_set = 1
-				else if( y == user.y )
-					if( x > user.x )
-						dir_to_set = 8
+						dir_to_set = 1 //User is laying from the top
+				else if(y == user.y) //User is on the same horizontal plane
+					if (x > user.x)
+						dir_to_set = 8 //User is laying from the left
 					else
-						dir_to_set = 4
+						dir_to_set = 4 //User is laying from the right
 			else
-				user << "<span class='notice'>You can't reach.</span>"
-				return //Only works for cardinal direcitons, diagonals aren't supposed to work like this.
-		for(var/obj/structure/window/WINDOW in loc)
-			if(WINDOW.dir == dir_to_set)
-				user << "<span class='notice'>There is already a window facing this way there.</span>"
+				user << "<span class='warning'>You can't reach far enough.</span>"
 				return
-		user << "<span class='notice'>You start placing the window.</span>"
-		if(do_after(user,20))
-			if(!src) return //Grille destroyed while waiting
-			for(var/obj/structure/window/WINDOW in loc)
-				if(WINDOW.dir == dir_to_set)//checking this for a 2nd time to check if a window was made while we were waiting.
-					user << "<span class='notice'>There is already a window facing this way there.</span>"
+		for(var/obj/structure/window/P in loc)
+			if(P.dir == dir_to_set)
+				user << "<span class='warning'>There's already a window here.</span>" //You idiot
+				return
+		user.visible_message("<span class='notice'>[user] starts placing a window on \the [src].</span>", \
+		"<span class='notice'>You start placing a window on \the [src].</span>")
+		if(do_after(user, 20))
+			for(var/obj/structure/window/P in loc)
+				if(P.dir == dir_to_set)//checking this for a 2nd time to check if a window was made while we were waiting.
+					user << "<span class='warning'>There's already a window here.</span>"
 					return
-			var/obj/item/stack/sheet/glass/glass/G = W
-			var/obj/structure/window/WD = new G.created_window(loc,0)
+			var/obj/item/stack/sheet/glass/glass/G = W //This fucking stacks code holy shit
+			var/obj/structure/window/WD = new G.created_window(loc, 0)
 			WD.dir = dir_to_set
 			WD.ini_dir = dir_to_set
 			WD.anchored = 0
 			WD.d_state = 0
-			var/obj/item/stack/ST = W
+			var/obj/item/stack/ST = W //HOLY FUCKING SHIT !
 			ST.use(1)
-			user << "<span class='notice'>You place the [WD] on [src].</span>"
+			user.visible_message("<span class='notice'>[user] places \a [WD] on \the [src].</span>", \
+			"<span class='notice'>You place \a [WD] on \the [src].</span>")
 		return
-//window placing end
 
 	else if(istype(W, /obj/item/weapon/shard))

```

Comment:
I mean, glass shards are intentionally nerfed because otherwise they cut through grilles in seconds. I kept that nerf, it has nothing to do with the item being conductive or not unless you want all non-conductive items to do 1/10th damage